redash/handlers/query_results.py

```diff
@@ -89,8 +43,15 @@ def run_query(query, parameters, data_source, query_id, max_age=0):
     if query_result:
         return {'query_result': query_result.to_dict()}
     else:
-        job = enqueue_query(query.text, data_source, current_user.id, metadata={
-            "Username": current_user.email,
+        if current_user.is_api_user:
+            id = None
+            name = repr(current_user)
+        else:
+            id = current_user.id
+            name = current_user.email
+
+        job = enqueue_query(query.text, data_source, id, metadata={
```

Comment:
1. `id` is a reserved word.
2. We use the user id for the Query Results query runner. I think we need to pass in information about the API key for everything to work smoothly.

Reply:
1. 2692aa1
2. That's solid. 6ceac2e should address this. I'd appreciate if you could give it a close look as it feels like a sensitive area. Also, are there any other query runners who need `user`?